libraries/profile.rb

```diff
@@ -141,6 +141,22 @@ class ComplianceProfile < Chef::Resource
         end
       end
 
+      action :upload do
+        converge_by 'upload compliance profile' do
+          raise 'unable to determine path to compressed profile archive' if path.nil?
+          o, p = normalize_owner_profile
+          node.run_state['compliance'] ||= {}
+
+          if node.run_state['compliance']['access_token']
+            reqpath ="owners/#{o}/compliance/#{p}/tar"
+            url = construct_url(server, reqpath)
+            Chef::Log.info "Upload compliance profile #{o}/#{p} to: #{url}"
+            res = Compliance::HTTP.post_file(url.to_s, node.run_state['compliance']['access_token'], path, insecure)
```

Comment:
`Compliance::API.upload()` calls `Compliance::HTTP.post_file()` like so:

`res = Compliance::HTTP.post_file(url, config['token'], archive_path, config['insecure'], !config.supported?(:oidc))`

However, when I pass in a config from `Compliance::Configuration.new()`, `!config.supported?(:oidc))` always resolves to `true` then authentication fails, when `false` the call succeeds.

```
pry(#<Audit::Resource::ComplianceProfile action provider>)> !config.supported?(:oidc)
=> true
[10] pry(#<Audit::Resource::ComplianceProfile action provider>)> !config.supported?(:bar)
=> true
[11] pry(#<Audit::Resource::ComplianceProfile action provider>)>
```